rfcs/0000-nixpkgs-ifd.md

```diff
@@ -0,0 +1,182 @@
+   Per the 3rd rule for Nixpkgs above, as long as we make the vendoring automatic and pure, this is fine, and improvement upon today.
+   Also, even if we didn't have the "one round of dynamism" restriction, we would still have the bootstrapping issue.
+
+# Alternatives
+[alternatives]: #alternatives
+
+1. Instead of evaluating Nixpkgs twice, just evaluate `allImportedDerivations` the first round.
+
+   We could do this, and would reduce total eval time, yes.
+   But, I think it would come at the cost of inciting great controversy.
+   This means users of the `enableImportFromDerivation = false` subset of Nixpkgs would still have to *wait*, for all the IFD to complete first.
+   And remember, with mass rebuilds, that could be quite some time.
+   Increasing the critical path length of *everything* we do with Nixpkgs would cause real pain in some quarters, and I don't want that to pay that as the cost of IFD.
+
+   With the plan as written, users of packages depending on IFD do have to wait slightly longer as the first eval is longer (and we wait for it before beginning to build `allImportedDerivations`).
+   But I think that is fair; we would be the "new constituency", the bottom of the pecking order, and so we should be patient so that other's workflows are not disturbed.
+
+   Longer term we could revisit this, or we could e.g. double down on automatic vendoring, committing all generated code to a second "roll-up" repo.
+   Many options between those two extremes; I rather not worry to much about it now and just take the conservative polite route proposed here to begin.
+
+2. As always, do nothing, and keep the status quo.
+
+# Unresolved questions
+[unresolved]: #unresolved-questions
+
+Should we call it "import from Derivation", or should we give it a different name?
+`builtins.readFile <drv>` is really the same thing for our purposes, so I am sympathetic to renaming.
+
+# Future work
+[future]: #future-work
+
+In a grand future we might do things completely differently.
+But I have no idea how stuff is going to shake out.
+In particular, if we don't do something like this, I don't think we will ever get to that future.
+So even if this technically "barking up the wrong tree", I think it is a necessary first step to get things going.
+
+I will say, though, with these steps, I think we will be able to successfully convert to Nix a bunch of developers that mainly work in one language, and didn't even think they were in need of a better distro.
+In turn, I hope these upstream packages and ecosystems might even care about packaging and integration of the sort that we do.
+This would create a virtuous cycle where Nix is easier to use by more people, and Nixpkgs is easier to maintain as upstream packages better match our values.
+
+The most important future work is technical, but being able to win upstream developer hearts and minds better than before, because ultimately distribution's live and die by upstream's decisions.
```

Comment:
`is technical` / `is not technical`

(I was paying attention :smile:)

Reply:
```suggestion
The most important future work is not technical, but being able to win upstream developer hearts and minds better than before, because ultimately distribution's live and die by upstream's decisions.
```